### DISCUSSION TITLE
[Feature] Sort shared albums by most recent assets added first in the list, in Sharing view and in any dialog that lists Shared albums

### DISCUSSION THREAD

**Discussion by jon-f-novabackup** [immich-app/immich#2319] [Feature Request]
I'm requesting that a new sort order be implemented that provides the ability to sort shared albums by most recent assets added first in the list, in Sharing view and in any dialog that lists Shared album, such as the list seen when attempting to add assets via Photos timeline when select the Shared album. I believe that it would be useful to have a way to show the order of shared albums by the most recent assets added, so that the shared albums in [Sharing] tab with the newest added assets are seen at the top of the albums list, first in the list. I think from a user experience standpoint it makes sense to at least provide the option to do this. Right now Immich, as far as date based sort order, only lists shared albums in [Sharing] tab by original date created, which is hard to realize about recent activity in albums, like to know which albums are active, etc. In google, web and mobile app, tested on 04/22/23, if any user adds a new asset to either a non-shared album or a shared album, it automatically within a few seconds moves that album to the top of the list as the first album that is listed in both the [Sharing] tab and the [Library] tab, and it executes this by default for both the Albums and Shared Albums pages in web and mobile. The logic in google is that it only takes adding at least 1 new asset to the album to trigger it to do that (move to top of the list for the sort order), but if that new asset is removed from the album, or any asset it removed from the album it doesn't trigger the move to top operation again, it stays as is for the sort order. If a new comment is added to the album it does trigger the move to the top however. If you like an asset it doesn't trigger the move to the top, nor does renaming the album. I think this logic makes sense to become the default method of albums sort by date order, because right now if you create albums or share albums they only get listed by date of the original album creation, however it would be useful especially for shared albums visible to many users, if any new assets are added to an album it would make sense to show that album as the first in the albums listing, as long as the sort order preference wasn't already using alphabetical / by album name sort order. It could either work hand in hand with the "Most recently created" sort order or add an additional 3rd sort order to the drop down menu, and it may even make sense to utilize that new method by default if the "By album title" (alpha) isn't the current sort order chosen.

Note: There is a similar issue to this, from June 2022, but it doesn't cover the Sharing view (mobile) or page (web), only the Albums view, and is only requested for web. If this request needs to be merged, or add to it the fact that this feature is also needed for the "Sharing view" (shared albums section in both web and mobile app), not just for "albums view" for web, as it is stated in # 1689, then go ahead. I'm not sure if it would be acceptable for me to add to that by commenting in itahead or inform me.

**Comment by alextran1502** [comment]
This is similar to request you posted yesterday in #2135 correct?

**Reply by jon-f-novabackup** [reply]
> This is similar to request you posted yesterday in #2135 correct?

I did not create that item #2135, but I did create a feature request #2315, to do with sort order, to get the existing sort order functionality that the mobile app contains in Library view implemented that same way in the Sharing view (it only exists in Albums view in mobile). This new request would be to mirror the feature that google does by default where if you add assets to any shared or otherwise album then it automatically lists those albums first in the listed order, in both web and mobile app, as the default sort order (without even choosing a sort order actually).
